kong/pdk/request.lua

```diff
@@ -279,12 +279,28 @@ local function new(self)
   function _REQUEST.get_path()
     check_phase(PHASES.request)
 
-    local uri = ngx.var.request_uri
+    local uri = ngx.var.request_uri or ""
     local s = find(uri, "?", 2, true)
     return s and sub(uri, 1, s - 1) or uri
   end
 
 
+  ---
+  -- When the request has a query string, returns the path + "?" + the query string.
+  -- Otherwise it returns just the path. No transformations/normalizations are done.
+  --
+  -- @function kong.request.get_path_and_querystring()
+  -- @phases rewrite, access, header_filter, body_filter, log
+  -- @treturn string the path and querystring
+  -- @usage
+  -- -- Given a request to https://example.com:1234/v1/movies?movie=foo
+  --
+  -- kong.request.get_path_and_querystring() -- "/v1/movies?movie=foo"
+  function _REQUEST.get_path_and_querystring()
```

Comment:
I left comment about the name here:
https://github.com/Kong/kong/pull/3859#pullrequestreview-165268435